cli/src/commands/git/push.rs

```diff
@@ -591,10 +575,8 @@ fn validate_commits_ready_to_push(
             }
             return Err(error);
         }
-        if let Some(sign_settings) = &sign_settings {
-            if !commit.is_signed() && sign_settings.should_sign(commit.store_commit()) {
-                commits_to_sign.push(commit);
-            }
+        if !commit.is_signed() && should_sign_on_push(commit.id())? {
```

Comment:
nit: I would drop `!commit.is_signed()` and set the default of `git.sign-on-push` to `~signed() && mine()`.

```suggestion
        if should_sign_on_push(commit.id())? {
```

Reply:
Not sure if this would make it more annoying for users to customize `git.sign-on-push`. I think it would be acceptable if it's well-documented though.